api/src/main/java/io/opentelemetry/metrics/AsynchronousInstrument.java

```diff
@@ -54,10 +54,12 @@
    */
   void setCallback(Callback<R> callback);
 
+  Observation observation(T observation);
```

Comment:
One bummer about making this generic is that it means that every number will get boxed with an object wrapper, rather than just using the primitive. Perhaps for Observers this is less of an issue than with synchronous instruments, but I do want to make sure we're ok with the extra allocations this will result in.

Reply:
Also, this means that users could pass in nulls, which we would need to account for. Is it possible to just have 2 primitive-only operations? I think we could get rid of the `getType()` method then as well, maybe?